gcloud-java-bigquery/README.md

```diff
@@ -36,14 +41,239 @@ About Google Cloud BigQuery
+the code from the main method to your application's servlet class and change the print statements to
+display on your webpage.
+
+```java
+import com.google.gcloud.bigquery.BaseTableInfo;
+import com.google.gcloud.bigquery.BigQuery;
+import com.google.gcloud.bigquery.BigQueryOptions;
+import com.google.gcloud.bigquery.DatasetInfo;
+import com.google.gcloud.bigquery.Field;
+import com.google.gcloud.bigquery.FieldValue;
+import com.google.gcloud.bigquery.InsertAllRequest;
+import com.google.gcloud.bigquery.InsertAllResponse;
+import com.google.gcloud.bigquery.QueryRequest;
+import com.google.gcloud.bigquery.QueryResponse;
+import com.google.gcloud.bigquery.Schema;
+import com.google.gcloud.bigquery.TableId;
+import com.google.gcloud.bigquery.TableInfo;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class GcloudBigQueryExample {
+
+  public static void main(String[] args) throws InterruptedException {
+
+    // Create a service instance
+    BigQuery bigquery = BigQueryOptions.defaultInstance().service();
+
+    // Create a dataset
+    String datasetId = "my_dataset_id";
+    bigquery.create(DatasetInfo.builder(datasetId).build());
+
+    TableId tableId = TableId.of(datasetId, "my_table_id");
+    // Table field definition
+    Field stringField = Field.builder("StringField", Field.Type.string())
+        .mode(Field.Mode.NULLABLE)
+        .build();
+    // Table schema definition
+    Schema schema = Schema.of(stringField);
+    // Create a table
+    BaseTableInfo createdTableInfo = bigquery.create(TableInfo.of(tableId, schema));
+
+    // Define rows to insert
+    Map<String, Object> firstRow = new HashMap<>();
+    Map<String, Object> secondRow = new HashMap<>();
+    firstRow.put("StringField", "value1");
+    secondRow.put("StringField", "value2");
+    // Create an insert request
+    InsertAllRequest insertRequest = InsertAllRequest.builder(tableId)
+        .addRow(firstRow)
+        .addRow(secondRow)
+        .build();
+    // Insert rows
+    InsertAllResponse insertResponse = bigquery.insertAll(insertRequest);
+    // Check if errors occurred
+    if (insertResponse.hasErrors()) {
+      System.out.println("Errors occurred while inserting rows");
+    }
+
+    // Create a query request
+    QueryRequest queryRequest = QueryRequest.builder("SELECT * FROM my_dataset_id.my_table_id")
+        .maxWaitTime(60000L)
+        .maxResults(1000L)
+        .build();
+    // Request query to be executed and wait for results
+    QueryResponse queryResponse = bigquery.query(queryRequest);
+    while (!queryResponse.jobComplete()) {
+      Thread.sleep(1000);
+      queryResponse = bigquery.getQueryResults(queryResponse.jobId());
+    }
+    // Read rows
+    Iterator<List<FieldValue>> rowIterator = queryResponse.result().iterateAll();
+    System.out.println("Table rows:");
+    while (rowIterator.hasNext()) {
+      System.out.println(rowIterator.next());
+    }
+  }
+}
+```
```

Comment:
We added a troubleshooting [section](https://github.com/GoogleCloudPlatform/gcloud-java/tree/master/gcloud-java-datastore#troubleshooting).